src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -1024,3 +1072,480 @@ def _read_semi_structured_file(self, path: str, format: str) -> DataFrame:
         df._reader = self
         set_api_call_source(df, f"DataFrameReader.{format.lower()}")
         return df
+
+    @private_preview(version="1.29.0")
+    def dbapi(
+        self,
+        create_connection: Callable[[], "Connection"],
+        table: str,
+        *,
+        column: Optional[str] = None,
+        lower_bound: Optional[Union[str, int]] = None,
+        upper_bound: Optional[Union[str, int]] = None,
+        num_partitions: Optional[int] = None,
+        max_workers: Optional[int] = None,
+        query_timeout: Optional[int] = 0,
+        fetch_size: Optional[int] = 0,
```

Comment:
I don't see insert batch size. Is it the same as fetch_size?

Reply:
we don't have insert batch size for our ingestion -- or you can assume it's the same as fetch_size.

in pyspark, batchsize is a parameters applied to write, not to read: https://spark.apache.org/docs/3.5.4/sql-data-sources-jdbc.html